Sources/Core/TypedProgram.swift

```diff
@@ -26,6 +26,11 @@ public struct TypedProgram: Program {
   /// A map from sequence expressions to their evaluation order.
   public let foldedSequenceExprs: [NodeID<SequenceExpr>: FoldedSequenceExpr]
 
+  /// The Val standard library module.
+  public var corelib: ModuleDecl.Typed? {
+    ast.corelib != nil ? self[ast.corelib!] : nil
```

Comment:
```suggestion
    ast.corelib == nil : nil ? self[ast.corelib!]
```
to avoid testing a negative. Or perhaps even better:
```suggestion
    ast.corelib.map({ self[$0] })
```